zeroex/orderwatch/order_watcher.go

```diff
@@ -1959,8 +1959,22 @@ func (w *Watcher) checkDecodeErr(err error, eventType string) bool {
 		}).Warn("unsupported event found")
 		return true
 	}
+	// NOTE(oskar): What happens here is that some tokens which do not
+	// respect the ERC20 specification can throw ABI decoder errors, we
+	// should handle them here and return it as a non-critical error.
+	// TODO(oskar): Should this be handled the same way for all non-ABI
+	// conforming ERC20 implementations?
```

Comment:
> Should this be handled the same way for all non-ABI  conforming ERC20 implementations?

Where the failures just ignored in the old behaviour? If so, the new behaviour just adds a log, which is backwards compatible in that if no-one had a problem with the old behaviour, they should be fine with the new behaviour too.